src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -962,6 +984,19 @@ private void GeneratePropertyGetter<T>(QualifiedType returnType, T decl, Class @
                     ReturnType = decl.QualifiedType
                 };
 
+                var arrayType = field.Type as ArrayType;
+
+                if (arrayType != null && @class.IsValueType)
+                {
+                    CSharpTypePrinter typePrinter = new CSharpTypePrinter(ctx.Driver);
+                    string type = arrayType.CSharpType(typePrinter).Type;
+                    WriteLine(string.Format("fixed ({0} __arrPtr = {1}.{2})",
```

Comment:
Also I saw that getters and setters are not generated for multidimentional arrays.
So, wouldn't it be good to simply use the substring of 'type' before last two characters as there will always be only one set of [].
This works perfectly fine too :
https://gist.github.com/genuinelucifer/45a29b36ec4f5254ea21